### DISCUSSION TITLE
Connection with mysql

### DISCUSSION THREAD

**Discussion by brunop26** [gama-platform/gama#210] [Q&A] [answered]
Hello,
I can't connect to my sql database with the following map, even though it works with python or java using the same map (I use a macbook pro) :
map <string, string> MySQL <- [
'host'::'localhost',
'dbtype'::'MySQL',
'database'::'pria',
'port'::'3306',
'user'::'pria',
'passwd'::'pria'];
I have the same problem when the map is on a single line. Do you have a solution?
Thank you,

**Comment by lesquoyb** [comment, marked as answer]
@benoitgaudou any clue ?

**Comment by brunop26** [comment]
Hello,
I have the same problem with postsgre. In short, I can't connect to MySQL or Postgres even though it works with java or python using the same map. Note that I have no problem with SQLite. I'm using a macbook pro OS sonoma 14.5. For the record, Benoit Goudou also seems to be experiencing this problem. 
Thank you

**Reply by lesquoyb** [reply]
Which version of gama are you using ?
Did it work before ? If so, with which version ?
Also is there any error message ?

**Comment by brunop26** [comment]
Hello,
It doesn't work with either version 1.9.2 or 1.9.3, which I'm currently using.

**Comment by benoitgaudou** [comment]
Dear all,

Basically the error seems to come from an issue or a change in the mysql
connector or on the mysql client:
it seems necessary to set explicity the time zone ...

A first dig and some tries seem to show a workaround outside of the GAMA or
GAML code.
In the MySQL client, it is possible to define a configuration file (my.cnf).
We can add to it:

[mysqld]
default_time_zone='+00:00'

After the change, close the client and relaunch everything.

As an example, using MAMP, you can refer to this documentation:
https://documentation-5.mamp.info/en/MAMP-Mac/How-Tos/Set-the-MySQL-Default-Storage-Engine/index.html

Hope it helps.
I am looking toward a solution in Java to avoid toi modify the MySQL server
....

But I am wondering whether we should provide a workaround in GAMA and
simply specify how to configure the server. Any thoughts?

Cheers
Benoit

**Comment by brunop26** [comment]
Hello,
I've just made the change and it's OK. Thanks a lot. The idea of making a bypass from Gama seems sensible to me.
